Penumbra/UI/SharedTagManager.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Dalamud.Interface;
+using ImGuiNET;
+using OtterGui;
+using OtterGui.Raii;
+using OtterGui.Widgets;
+using Penumbra.Mods;
+using Penumbra.UI.Classes;
+
+namespace Penumbra.UI;
+public sealed class SharedTagManager
+{
+    private static uint _tagButtonAddColor = ColorId.SharedTagAdd.Value();
+    private static uint _tagButtonRemoveColor = ColorId.SharedTagRemove.Value();
+
+    private static float _minTagButtonWidth = 15;
+
+    private const string PopupContext = "SharedTagsPopup";
+    private bool _isPopupOpen = false;
+
+
+    public IReadOnlyList<string> SharedTags { get; internal set; } = Array.Empty<string>();
```

Comment:
When removing the Tag list from config, it does not make a lot of sense to use an array here, instead of a list. 
The correct way here would be to have a 
```cs
private readonly List<string> _sharedTags = [];
public IReadOnlyList<string> SharedTags
    => _sharedTags;
```
inside this manager, and have the manager implement the ISavable.